### PR TITLE
Lpd 22505 master

### PR DESCRIPTION
**Pull request by SamZiemer**
https://liferay.atlassian.net/browse/LPD-22505

When a journal article is used in a web content display, and the DDMTemplate is specifically linked orphaned data is left behind upon deletion which prevents the template from being removed.

I've [submitted this PR](https://github.com/liferay-content-management/liferay-portal/pull/5321) to @liferay-content-management already and changes to JournalArticleLocalServiceImpl have been approved. I was requested to send here as well due to the upgrade process. Please let me know if there is anything else needed for this.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_4_5/DDMTemplateLinkUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		_classNameLocalService = classNameLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		String compositeClassName = ResourceActionsUtil.getCompositeModelName(
+			JournalArticle.class.getName(), DDMTemplate.class.getName());
+
+		runSQL(
+			"delete from ddmtemplatelink where classNameId = '" +
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_4_5/DDMTemplateLinkUpgradeProcess.java:31]: @SamZiemer please change it to `delete from DDMTemplateLink where classNameId = '`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_4_5/DDMTemplateLinkUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+				_classNameLocalService.getClassNameId(compositeClassName) +
+					"' AND classPK not in (select id_ from journalarticle)");
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_4_5/DDMTemplateLinkUpgradeProcess.java:33]: @SamZiemer please change it to `' and classPK not in (select id_ from JournalArticle)`

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by SamZiemer** [comment]
Hi @carolmariaabb, thank you for taking a look at this PR. I've added a commit making the changes you've requested

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**